src/lm/kaldi-lmtable.cc

```diff
@@ -207,15 +208,24 @@ bool LmTable::ReadFstFromLmFile(std::istream &istrm,
     }
     // found valid line
     ngram_order = atoi(inpline.substr(pos1+5, pos2-(pos1+5)).c_str());
+    orders.push_back(ngram_order);
     if (ngram_order > max_ngram_order) {
       max_ngram_order = ngram_order;
     }
   }
+  std::sort(orders.begin(), orders.end());
+  
   if (max_ngram_order == 0) {
     // reached end of loop without having found any n-gram
     KALDI_ERR << "No ngrams found in specified file";
   }
 
+  if (orders[0] != 1)
+    KALDI_ERR << "arpa file has no 1-grams specified";
+  for (std::vector<int>::iterator i = orders.begin()+1; i != orders.end(); ++i)
```

Comment:
would be easier and more readable to do
 for (int32 i = 0; i < orders.size(); i++) {
   if (orders[i] != i+1) { ....